nightly build fails on ubuntu 18.04 with unrecognized command line option "-Wmissing"

Whenever the source tree is updated the Ubuntu nightly packages are rebuilt (the "develop" repository).
As of yesterday night the build for Ubuntu bionic (18.04) fails with the below error message. It looks to me as if it is failing while cmake is probing (like autoconf).
The log file says it's using this version of gcc: `gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04) `
On the more modern Ubuntu versions (20.04 and 22.04) GTSAM builds just fine.

```
Source file was:
int main() { return 0; }
	cd obj-x86_64-linux-gnu && tail -v -n \+0 CMakeFiles/CMakeError.log
==> CMakeFiles/CMakeError.log <==
Performing C++ SOURCE FILE Test COMPILER_HAS_WMISSING_OVERRIDE failed with the following output:
Change Dir: /<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_3e948/fast"
make[2]: Entering directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
/usr/bin/make -f CMakeFiles/cmTC_3e948.dir/build.make CMakeFiles/cmTC_3e948.dir/build
make[3]: Entering directory '/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_3e948.dir/src.cxx.o
/usr/bin/c++    -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -DCOMPILER_HAS_WMISSING_OVERRIDE   -Wmissing -o CMakeFiles/cmTC_3e948.dir/src.cxx.o -c /<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/src.cxx
c++: error: unrecognized command line option '-Wmissing'; did you mean '-Waliasing'?
CMakeFiles/cmTC_3e948.dir/build.make:65: recipe for target 'CMakeFiles/cmTC_3e948.dir/src.cxx.o' failed
make[3]: *** [CMakeFiles/cmTC_3e948.dir/src.cxx.o] Error 1

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,7 +54,7 @@ include(cmake/HandleGeneralOptions.cmake)   # CMake build options
 # Enable or disable serialization with GTSAM_ENABLE_BOOST_SERIALIZATION
 option(GTSAM_ENABLE_BOOST_SERIALIZATION "Enable Boost serialization" ON)
 if(GTSAM_ENABLE_BOOST_SERIALIZATION)
-  add_compile_definitions(GTSAM_ENABLE_BOOST_SERIALIZATION)
+  add_definitions(-DGTSAM_ENABLE_BOOST_SERIALIZATION)
 endif()
 
 option(GTSAM_USE_BOOST_FEATURES "Enable Features that use Boost" ON)

